### PR TITLE
Create Mesos client.

### PR DESCRIPTION
**Pull request by jeschkies**
The connect method handles the request to Mesos' event stream.
It extracts the framework ID from the first event and captures the
stream id. These are required for future request to the API.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -185,12 +188,117 @@ impl Stream for Events {
...
+        call.set_subscribe(subscribe);
+        call.set_field_type(scheduler::Call_Type::SUBSCRIBE);
+        call
+    }
+
+    /// Builds a Hyper Request object for given URI and Mesos scheduler call.
+    pub fn request_for(uri: hyper::Uri, call: scheduler::Call) -> hyper::Request {
+        let mut request = hyper::Request::new(hyper::Method::Post, uri);
+        // TODO: construct media type once
+        let protobuf_media_type = "application/x-protobuf".parse::<mime::Mime>().unwrap();
```
Comment (boxdot) [inline src/client.rs]: lazy_static is your friend

File: src/client.rs [hunk 2]
```diff
@@ -185,12 +188,117 @@ impl Stream for Events {
...
+    ) -> Result<Self, Error> {
+        if let Some(event) = maybe_event {
+            if event.has_subscribed() {
+                let framework_id = event
+                    .get_subscribed()
+                    .get_framework_id()
+                    .clone();
+
+                Ok(Self {
+                    framework_id: String::from(framework_id.get_value()),
```
Comment (boxdot) [inline src/client.rs]: JFYI: I usually use a shorter version `framework_id.get_value.into()`. But this is also not explicit.

File: src/client.rs [hunk 3]
```diff
@@ -185,12 +188,117 @@ impl Stream for Events {
...
+        let call = Self::subscribe(framework_info);
+        let request = Self::request_for(uri, call);
+
+        // Call Mesos
+        let http_client = hyper::Client::new(&handle);
+        let s = http_client
+            .request(request)
+            .map_err(failure::Error::from)
+            .and_then(move |res: hyper::Response| {
+                println!("Response status: {}", res.status());
```
Comment (boxdot) [inline src/client.rs]: https://docs.rs/log/0.4.1/log/ especially since this is a library

File: src/client.rs [hunk 4]
```diff
@@ -185,12 +188,117 @@ impl Stream for Events {
...
+        ));
+
+        // TODO: Handle error
+        let body = call.write_to_bytes().unwrap();
+        request.set_body(body);
+        request
+    }
+
+    /// Builds a new Mesos client from subcribe event, remaining Mesos events and Mesos streamd id.
+    fn client_from(
```
Comment (boxdot) [inline src/client.rs]: Usually such methods are called just `new` or `with_...`.

File: tests/mesos.rs [hunk 5]
```diff
@@ -9,62 +9,37 @@ extern crate tokio_core;
...
-        // Fire off request
-        let work = client.request(request).map(|res| {
-            println!("Response status: {}", res.status());
-            return Events::new(res.body());
-        });
-
-        let events: Events = core.run(work).unwrap();
-        let w = events.map(|event| event.get_field_type()).take(2).collect();
+        let events: Events = core.run(client).unwrap().events;
+        let w = events.map(|event| event.get_field_type()).take(1).collect();
```
Comment (boxdot) [inline tests/mesos.rs]: Just `map(scheduler::Event::get_field_type)`?

**Comment by boxdot** [comment]
Looks good!

File: tests/mesos.rs (hunk 5 above)
Comment (jeschkies) [reply tests/mesos.rs]: I don't think that works.

File: src/client.rs (hunk 4 above)
Comment (jeschkies) [reply src/client.rs]: Done

File: src/client.rs (hunk 2 above)
Comment (jeschkies) [reply src/client.rs]: Done

File: src/client.rs (hunk 1 above)
Comment (jeschkies) [reply src/client.rs]: Done. At some point it should just be a `const mime::Mime` such as `mime::Text_PLAIN`

File: src/client.rs (hunk 3 above)
Comment (jeschkies) [reply src/client.rs]: Done.

**Review by boxdot — APPROVED** [review]

**Pull request merged**